code/asteroid/asteroid.cpp

```diff
@@ -798,9 +806,14 @@ static void maybe_throw_asteroid()
 		target.throw_stamp = timestamp(1000 + 1200 * target.incoming_asteroids /(Game_skill_level+1));
 
 		int subtype = Random::next(NUM_DEBRIS_POFS);
```

Comment:
Hmm.  If there are fewer than `NUM_DEBRIS_POFS`, this method of choosing an asteroid will bias towards choosing the first one.  For 3 max POFs and only two types, the first type will be chosen 2/3rds of the time.  Can you write a new function to just count the number of valid subtypes and choose a random index from that count?  Use it both here and in the other location.